includes/discount-functions.php

```diff
@@ -32,19 +32,29 @@ function edd_get_discounts( $args = array() ) {
 
 	$args = wp_parse_args( $args, $defaults );
 
-	$discounts = get_posts( $args );
+	$discounts_hash = md5( json_encode( $args ) );
+	if ( ! $discounts = wp_cache_get( $discounts_hash, 'edd-discounts' ) ) {
```

Comment:
`if ( ! $discounts = wp_cache_get( $discounts_hash, 'edd-discounts' ) ) {` 
is valid for php 5.3 (verified), but I can understand if this is not a preferred method. Can you confirm if this needs changing either way?